_posts/en/newsletters/2025-03-14-newsletter.md

```diff
@@ -0,0 +1,157 @@
+  Delving Bitcoin an analysis of the network traffic generated and
+  received by his node in four different modes: initial block download
+  (IBD), non-listening, non-archival (pruned) listening, and archival
+  listening.  Although the results for his single node may not have been
+  representative in all cases, we found several of his discoveries to by
+  interesting:
+
+  - *High block traffic as archival listening node:* Virtu's node served
+    several gigabytes of blocks each hour to other nodes when operating
+    as a non-pruned listening node.  Many of blocks were older blocks
+    being requested by incoming connections so that they could perform
+    IBD.
+
+  - *High inv traffic as non-archival listener:* about 20% of total node
+    traffic was `inv` messages before he enabled serving of older
+    blocks.  [Erlay][topic erlay] might be able to significantly reduce
+    that 20% overhead, which represented about 100 megabytes a day.
+
+  - *Bulk of inbound peers appear to be spy nodes:* "Interestingly, the
+    bulk of inbound peers exchange only around 1MB of traffic with my
+    node, which is too low (using traffic via my outbound connections as
+    baseline) for them to be regular connections. All those nodes do is
+    complete the P2P handshake, and politely reply to ping messages.
+    Other than that, they just suck up our `inv` messages."
+
+  Virtu's post contains additional insights and multiple charts
+  illustrating the traffic experienced by his node.
+
+- **Research into single-path LN pathfinding:** Sindura Saraswathi
+  [posted][saraswathi path] to Delving Bitcoin about [research][sk path]
+  she conducted with Christian Kümmerle about finding optimal paths
+  between LN nodes for sending payments in a single part.  Her post
+  describes the strategies currently used by Core Lightning, Eclair,
+  LDK, and LND.  The authors then use eight modified and unmodified LN
+  nodes in a simulated LN network (based on a snapshot of the actual
+  network) to test pathfinding, evaluating based criteria such as
+  highest success rate, lowest fee ratios (lowest cost), shortest total
+  time lock (least bad worst-case waiting period), and shortest path
+  (least likely to result in a stuck payment).  No algorithm performed
+  better than the others in all cases, and Saraswathi suggests that
+  implementations provide better weighting functions that allow users to
+  choose the tradeoffs they prefer for different payments (e.g., you may
+  prioritize a high success rate for a small in-person purchase but
+  prefer a low fee ratio for paying a large monthly bill that isn't due
+  for a few more weeks).  She also notes that "[although]  beyond the
+  scope of this study, we note that the insights gained in this study
+  are also relevant for future improvements in [multi-part
+  payment][topic multipath payments] pathfinding algorithms."
+
+- **Probabilistic payments using different hash functions as an xor function:**
```

Comment:
```suggestion
- **Probabilistic payments using different hash functions as an XOR function:**
```

Reply:
Keeping lower case.  It does appear in the dictionary and Wikipedia in all caps, but I think excessive use of capital letters is distracting and that we should avoid it when it doesn't disrupt meaning or our other obligations to readers.